test/specs/events.spec.js

```diff
@@ -1,56 +1,224 @@
+            delete targetOptions.click;
+            done();
+          });
+        });
+      });
+
+      it('should not detect any events (because unmanaged event)', function(done) {
+        const clickSpy = jasmine.createSpy('click');
+
+        targetOptions.click = clickSpy;
+        pluginOpts.annotations = [options];
 
+        const chart = window.acquireChart(chartConfig);
+        window.afterEvent(chart, 'touchstart', function() {
+          expect(clickSpy.calls.count()).toBe(0);
+          delete targetOptions.click;
+          done();
+        });
+        window.triggerMouseEvent(chart, 'touchstart', center(chart));
+      });
+
+      it('should not call removed hook', function(done) {
+        const enterSpy = jasmine.createSpy('enter');
+        const leaveSpy = jasmine.createSpy('leave');
+
+        pluginOpts.enter = enterSpy;
+        pluginOpts.leave = leaveSpy;
+        pluginOpts.annotations = [options];
+
+        const chart = window.acquireChart(chartConfig);
+        pluginOpts.enter = undefined;
+        chart.update();
+
+        window.triggerMouseEvent(chart, 'mousemove', center(chart));
+        window.afterEvent(chart, 'mousemove', function() {
+          expect(enterSpy.calls.count()).toBe(0);
+
+          window.triggerMouseEvent(chart, 'mousemove', x0y0);
+
+          window.afterEvent(chart, 'mousemove', function() {
+            expect(leaveSpy.calls.count()).toBe(1);
+            delete pluginOpts.enter;
+            delete pluginOpts.leave;
+            done();
+          });
+        });
+      });
+
+      it('should detect a property in the context, to check persistency', function(done) {
+        targetOptions.enter = function(ctx) {
+          ctx.persistency = true;
+        };
+        targetOptions.leave = function(ctx) {
+          expect(ctx.persistency).toBe(true);
+          done();
+        };
+        pluginOpts.annotations = [options];
+
+        const chart = window.acquireChart(chartConfig);
+        window.triggerMouseEvent(chart, 'mousemove', center(chart));
+        window.afterEvent(chart, 'mousemove', function() {
+
+          window.triggerMouseEvent(chart, 'mousemove', x0y0);
+
+          window.afterEvent(chart, 'mousemove', function() {
+            delete targetOptions.enter;
+            delete targetOptions.leave;
+          });
+        });
+      });
+
+      it('should detect a click event even if 2 clicks are fired', function(done) {
+        const dblClickSpy = jasmine.createSpy('dblclick');
+
+        targetOptions.dblclick = dblClickSpy;
+        pluginOpts.dblClickSpeed = 1;
+        pluginOpts.annotations = [options];
+
+        const chart = window.acquireChart(chartConfig);
+        const eventPoint = center(chart);
+
+        let dblClick = false;
+        window.afterEvent(chart, 'click', function() {
+          if (!dblClick) {
+            dblClick = true;
+            setTimeout(() => {
+              window.triggerMouseEvent(chart, 'click', eventPoint);
```

Comment:
I'm not sure why this completes, because `afterEvent` only listens for one event. Anyway, its been like for a long time, so lets leave it as is for now.

Reply:
uhm... I didn't go in deep of afterEvent but I have the feeling that is catching more than 1 event. Maybe I'm wrong